### PR TITLE
Task/filter addon by project and environment

### PR DESCRIPTION
**Pull request by chriswk**
This adds filtering on projects and environments to frontend and goes together with [PR 1801 in unleash/unleash](https://github.com/Unleash/unleash/pull/1801)

I do believe we might want to add some explanation before the input fields to make it clearer that adding projects/environments narrows the amount of events your addon will be notified about.

### CODE REVIEW

File: src/component/addons/AddonForm/AddonForm.tsx [hunk 1]
```diff
@@ -43,21 +37,36 @@ interface IAddonFormProps {
...
 
 export const AddonForm: VFC<IAddonFormProps> = ({
-    editMode,
-    provider,
-    addon: initialValues,
-    fetch,
-}) => {
-    const { createAddon, updateAddon } = useAddonsApi();
-    const { setToastData, setToastApiError } = useToast();
+                                                    editMode,
```
[the remark this answers is not in the document]
Comment (chriswk) [reply src/component/addons/AddonForm/AddonForm.tsx]: But there's no prettier configuration in the project, only ESLint?

File: src/component/addons/AddonForm/AddonForm.styles [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import { makeStyles } from 'tss-react/mui';
```
[the remark this answers is not in the document]
Comment (chriswk) [reply src/component/addons/AddonForm/AddonForm.styles]: Fixed, we didn't need this styles file, but I guess

File: src/component/addons/AddonForm/AddonForm.styles [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            minWidth: 'auto',
+        },
+    },
+    buttonContainer: {
+        marginTop: 'auto',
+        display: 'flex',
+        justifyContent: 'flex-end',
+    },
+    cancelButton: {
+        marginLeft: '1.5rem',
```
[the remark this answers is not in the document]
Comment (chriswk) [reply src/component/addons/AddonForm/AddonForm.styles]: Fixed by deleting a not needed file. Same goes here though, the place where I copied this from needs to be updated as well :)

File: src/component/addons/AddonForm/AddonForm.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (chriswk) [reply src/component/addons/AddonForm/AddonForm.tsx]: I formatted it manually for now

[the remark this answers is not in the document]
Comment (chriswk) [reply src/component/addons/AddonForm/AddonForm.tsx]: Ah, sorry, I see we have a fmt:check as well as the normal lint. Will use.

File: src/component/addons/AddonForm/AddonMultiSelector/AddonMultiSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+                    <SelectAllButton
+                        isAllSelected={isAllSelected}
+                        onClick={onSelectAllClick}
+                    />
+                }
+            />
+            {children}
+        </Fragment>
+    );
+    const CustomPaper = ({ ...props }) => <Paper elevation={8} {...props} />;
```
Comment (FredrikOseberg) [inline src/component/addons/AddonForm/AddonMultiSelector/AddonMultiSelector.tsx:116]: We should avoid creating react components inside of react components, this pattern may lead to unexpected bugs, especially with dynamic content like input fields and checkboxes.

**Review by FredrikOseberg — CHANGES_REQUESTED** [review]

File: src/component/addons/AddonForm/AddonMultiSelector/AddonMultiSelector.tsx [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+                }
+                label={`ALL current and future ${entityName}s`}
+            />
+        </Box>
+    );
+
+    const HelpText = () => (
+        <p>
+            Selecting {entityName}(s) here will filter events so that your addon
+            will only receive events that is tagged with one of your{' '}
```
Comment (sighphyre) [inline src/component/addons/AddonForm/AddonMultiSelector/AddonMultiSelector.tsx:137]:
```suggestion
            will only receive events that are tagged with one of your{' '}
```

**Review by sighphyre — COMMENTED** [review]
LGTM but I'd let the front end ninjas take one last glance here

**Review by sighphyre — APPROVED** [review]
LGTM!

**Review by FredrikOseberg — APPROVED** [review]
LG!

**Pull request merged**